app/controllers/books_controller.rb

```diff
@@ -1,18 +1,15 @@
 class BooksController < ApplicationController
-  before_action :set_book, only: [:show, :update, :destroy]
+  before_action :set_book, only: %i[show update destroy]
 
-  # GET /books
   def index
-    @books = Book.limit(params[:limit])
+    @books = Book.limit(params[:limit]).includes(:author, :publisher)
```

Comment:
I have included the author and the publisher here to avoid any N + 1 queries in cause we decided to use those in the Book serializer.